.github/workflows/ci_integration.yml

```diff
@@ -0,0 +1,129 @@
+# shellcheck disable=SC1000-SC9999
+name: Continuous Integration
+
+on:
+  workflow_call:
+    secrets:
+      AWS_REGION:
+        required: true
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+      AWS_SECURITY_GROUP_ID:
+        required: true
+      AMI_ID:
+        required: true
+      GH_SELF_HOSTED_RUNNER_TOKEN:
+        required: true
+
+permissions:
+  contents: read
+
+jobs:
+  setup-runner:
+    name: GitHub Actions Test create instance
+    runs-on: ubuntu-latest
+    outputs:
+      instance_id: ${{ steps.create-runner.outputs.instance_id }}
+      runner_name: ${{ steps.create-runner.outputs.runner_name }}
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v4
+
+      - name: Test Local Action
+        id: create-runner
+        uses: ./
+        with:
+            action: "create"
+            aws_region: ${{ secrets.AWS_REGION }}
+            github_token: ${{ secrets.GH_SELF_HOSTED_RUNNER_TOKEN }}
+            aws_access_key_id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+            aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+            security_group_id: ${{ secrets.AWS_SECURITY_GROUP_ID }}
+            github_repo: ${{ github.repository }}
+            ami_id: ${{ secrets.AMI_ID }}
+            instance_type: "t2.micro"
+            create_s3_bucket: "false"
+            spot_instance_only: "true"
+
+      - name: Print Output
+        id: output
+        run: |
+          echo "instance_id ${{ steps.create-runner.outputs.instance_id }}"
+          echo "instance_ip ${{ steps.create-runner.outputs.instance_ip }}"
+          echo "runner_name ${{ steps.create-runner.outputs.runner_name }}"
+          echo "bucket_name ${{ steps.create-runner.outputs.bucket_name }}"
+
+  test-runner:
+    needs: setup-runner
+    runs-on: [self-hosted, linux, x64]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install Docker
+        id: docker
+        run: |
+            # Add Docker's official GPG key:
```

Comment:
add a test here to see which OS the script is running on. if it is RHEL, use yum.

Reply:
as current our action with lots for apt command for ubuntu, I suppose for kepler action supports RHEL needs another topic, also for podman or runc...